### PR TITLE
Updates and fixes

### PR DESCRIPTION
**Pull request by Alexsandruss**
## Description

Changes:
- Add support of LightGBM daal4py modelbuilders
- Add garbage collection and result cleaning in datasets prefetching function to avoid out-of-memory errors
- Add `SKLBENCH_DATA_CACHE` env variable as the first default location for datasets cache for convenience (`$PWD/data_cache` is still working if env variable is not set)
- Change default dtype to float32
- Adjust compatibility mode of report generator to work with latest versions of stock sklearn and RAPIDS and for other cases
- Update collected performance metrics:
  - Add `cost` metrics counted in `microdollars` (most readable degree for usual case computation time)
  - Add CPU load profiling
  - Add RAM and VRAM usage profiling
  - Add coefficient of variation for time
  - Add 1st run time
  - Add 1st-mean run ratio
- Change color scale from RED-YELLOW-GREEN to RED-WHITE-GREEN in perf. report for better readability
- Add option for cache flushing between case runs
- Docs:
  - Mention Kaggle dataset download requirements
  - Note about experimental configs content and meaning
  - Move Benchmarking Config Specification to separate file
  - Add benchmarking scopes short explanation

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [ ] I have added a respective label(s) to PR if I have a permission for that.
- [ ] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [ ] I have run it locally and tested the changes extensively.
- [ ] All CI jobs are green or I have provided justification why they aren't.
- [ ] I have extended testing suite if new functionality was introduced in this PR.

### CODE REVIEW

File: configs/BENCH-CONFIG-SPEC.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+## Benchmark-Specific Parameters
+
+### `Scikit-learn Estimator`
+
+| Parameter keys | Default value | Choices | Description |
+|:---------------|:--------------|:--------|:------------|
+| `algorithm`:`estimator` | None |  | Name of measured estimator. |
+| `algorithm`:`estimator_params` | Empty `dict` |  | Parameters for estimator constructor. |
+| `algorithm`:`online_inference_mode` | False |  | Enables online mode for inference methods of estimator (separate call for each sample). |
```
Comment (david-cortes-intel) [inline configs/BENCH-CONFIG-SPEC.md:111]: Could mention whether those are parallelized or not (shouldn't be).

File: sklbench/datasets/README.md [hunk 2]
```diff
@@ -10,9 +10,13 @@ Data handling steps:
 Existing data sources:
  - Synthetic data from sklearn
  - OpenML datasets
+ - Kaggle competition datasets
```
Comment (david-cortes-intel) [inline sklbench/datasets/README.md:13]: Unrelated to these changes but: this repository would be a lot easier to use if it could avoid pulling data from kaggle.

File: sklbench/emulators/svs/neighbors.py [hunk 3]
```diff
@@ -42,15 +42,15 @@ def __init__(
         self.n_jobs = n_jobs
 
     def fit(self, X, y=None):
-        build_params = pysvs.VamanaBuildParameters(
+        build_params = svs.VamanaBuildParameters(
             graph_max_degree=self.graph_max_degree,
             window_size=self.window_size,
-            num_threads=self.n_jobs,
+            # num_threads=self.n_jobs,
```
[the remark this answers is not in the document]
Comment (Alexsandruss) [reply sklbench/emulators/svs/neighbors.py]: Done.
